plugins/include/json.inc

```diff
@@ -0,0 +1,788 @@
+// vim: set ts=4 sw=4 tw=99 noet:
+//
+// AMX Mod X, based on AMX Mod by Aleksander Naszko ("OLO").
+// Copyright (C) The AMX Mod X Development Team.
+//
+// This software is licensed under the GNU General Public License, version 3 or higher.
+// Additional exceptions apply. For full license details, see LICENSE.txt or visit:
+//     https://alliedmods.net/amxmodx-license
+
+//
+// JSON Natives
+//
+
+#if defined _json_included
+	#endinput
+#endif
+#define _json_included
+
+#pragma reqlib json
+#if !defined AMXMODX_NOAUTOLOAD
+	#pragma loadlib json
+#endif
+
+/*
+ * JSON types
+ */
+enum JSONType
+{
+	JSONError   = -1,
+	JSONNull    = 1,
+	JSONString  = 2,
+	JSONNumber  = 3,
+	JSONObject  = 4,
+	JSONArray   = 5,
+	JSONBoolean = 6
+};
+
+/*
+ * JSON invalid handle
+ */
+enum JSON
+{
+	Invalid_JSON = -1
+}
+
+/*
+ * Overloaded operators for convenient comparing
+ */
+native bool:operator==(const JSON:value1, const JSON:value2) = json_equals;
```

Comment:
I know I initially suggested this, but after putting more thought into it, I'd actually get rid of it.
While it's a cool feature, no other handle types in AMXX override `operator==`. People are used to `operator==` comparing handles/references, not their data, so an implicit deep comparison of JSON objects may be a bad idea.

Reply:
I agree as well, deep comparison should be explicit.

Reply:
Done.